### PR TITLE
Fix ArrayIndexOutOfBounds failure found in RTC 290630 test failure. A…

### PR DESCRIPTION
**Pull request by jvanhill**
…lso updating Infinispan version and adding some more checking on deactivate in CacheManagerService and CacheService.

### CODE REVIEW

**Review by ryanesch — APPROVED** [review]

**Comment by jvanhill** [comment]
#build

File: dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops [hunk 1]
```diff
@@ -72,10 +72,10 @@ CWLJC0011_GET_CACHE_ERR=CWLJC0011W: Error encountered while retrieving the {0} J
 CWLJC0011_GET_CACHE_ERR.explanation=The cache was not retrievable due to the specified error.
 CWLJC0011_GET_CACHE_ERR.useraction=Address the error included in the message and retry.
 
-CWLJC0012_CLOSE_CACHE_ERR=CWLJC0012E: Error closing the {0} JCache: {1}.
+CWLJC0012_CLOSE_CACHE_ERR=CWLJC0012W: Error closing the {0} JCache: {1}.
```
Comment (dmuelle) [inline dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]: is it possible to say what process or entity encountered the unspecified error? eg, 

`<x> encountered the following error when it tried to close the {0} JCache:`

Comment (dmuelle) [inline dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]:
```suggestion
CWLJC0012_CLOSE_CACHE_ERR=CWLJC0012W: The following error occurred when the server tried to close the {0} JCache: {1}.
```

File: dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops [hunk 2]
```diff
@@ -72,10 +72,10 @@ CWLJC0011_GET_CACHE_ERR=CWLJC0011W: Error encountered while retrieving the {0} J
...
 CWLJC0012_CLOSE_CACHE_ERR.explanation=Failed to close the JCache due to the specified error.
 CWLJC0012_CLOSE_CACHE_ERR.useraction=If this occurs repeatedly during runtime, it may result in leaked cache resources.
 
-CWLJC0013_CLOSE_CACHEMGR_ERR=CWLJC0013E: Error closing the {0} JCache CacheManager: {1}.
+CWLJC0013_CLOSE_CACHEMGR_ERR=CWLJC0013W: Error closing the {0} JCache CacheManager: {1}.
```
Comment (dmuelle) [inline dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]:
```suggestion
CWLJC0013_CLOSE_CACHEMGR_ERR=CWLJC0013W: The following error occurred when the server tried to close the {0} JCache CacheManager: {1}.
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
updated ID suggestions

File: dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops (hunk 1 above)
Comment (jvanhill) [reply dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]: See suggestions below.

Comment (jvanhill) [reply dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]: Agree to make these changes.

File: dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops (hunk 2 above)
Comment (jvanhill) [reply dev/io.openliberty.jcache.internal/resources/io.openliberty.jcache.internal/resources/JCacheMessages.nlsprops]: Agree to make these changes.

**Review by dmuelle — APPROVED** [review]
LGTM, thanks

**Comment by jvanhill** [comment]
#build

**Comment by jvanhill** [comment]
#libby

**Pull request merged**